utils/hct/hctdb.py

```diff
@@ -2181,7 +2187,13 @@ def UFI(name, **mappings):
             counters=('tex_cmp',))
         next_op_idx += 1
 
+        self.add_dxil_op("StartVertexLocation", next_op_idx, "StartVertexLocation", "returns the BaseVertexLocation from DrawIndexedInstanced or StartVertexLocation from DrawInstanced", "i", "rn", [
```

Comment:
--- hctdb.py	2023-10-27 02:11:34.000000 +0000
+++ hctdb.py	2023-10-27 02:13:38.534621 +0000
@@ -493,12 +493,11 @@
             self.name_idx[i].shader_model = 6,8
 
         for i in "StartVertexLocation,StartInstanceLocation".split(","):
             self.name_idx[i].category = "Extended Command Information"
             self.name_idx[i].shader_stages = ("vertex",)
-            self.name_idx[i].shader_model = 6,8
-
+            self.name_idx[i].shader_model = 6, 8
 
     def populate_llvm_instructions(self):
         # Add instructions that map to LLVM instructions.
         # This is basically include\llvm\IR\Instruction.def
         #
@@ -2185,16 +2184,30 @@
             db_dxil_param(12, "f", "bias", "bias value"),
             db_dxil_param(13, "f", "clamp", "clamp value")],
             counters=('tex_cmp',))
         next_op_idx += 1
 
-        self.add_dxil_op("StartVertexLocation", next_op_idx, "StartVertexLocation", "returns the BaseVertexLocation from DrawIndexedInstanced or StartVertexLocation from DrawInstanced", "i", "rn", [
-            db_dxil_param(0, "i32", "", "result")])
-        next_op_idx += 1
-
-        self.add_dxil_op("StartInstanceLocation", next_op_idx, "StartInstanceLocation", "returns the StartInstanceLocation from Draw*Instanced", "i", "rn", [
-            db_dxil_param(0, "i32", "", "result")])
+        self.add_dxil_op(
+            "StartVertexLocation",
+            next_op_idx,
+            "StartVertexLocation",
+            "returns the BaseVertexLocation from DrawIndexedInstanced or StartVertexLocation from DrawInstanced",
+            "i",
+            "rn",
+            [db_dxil_param(0, "i32", "", "result")],
+        )
+        next_op_idx += 1
+
+        self.add_dxil_op(
+            "StartInstanceLocation",
+            next_op_idx,
+            "StartInstanceLocation",
+            "returns the StartInstanceLocation from Draw*Instanced",
+            "i",
+            "rn",
+            [db_dxil_param(0, "i32", "", "result")],
+        )
         next_op_idx += 1
 
         # Set interesting properties.
         self.build_indices()
         for i in "CalculateLOD,DerivCoarseX,DerivCoarseY,DerivFineX,DerivFineY,Sample,SampleBias,SampleCmp".split(","):
@@ -2578,46 +2591,50 @@
                 if not anarg.is_ctor_param:
                     p_ids.add(anarg.name)
                 self.pass_idx_args.add(anarg.name)
 
     def build_semantics(self):
-        SemanticKind = db_dxil_enum("SemanticKind", "Semantic kind; Arbitrary or specific system value.", [
-            (0, "Arbitrary", ""),
-            (1, "VertexID", ""),
-            (2, "InstanceID", ""),
-            (3, "Position", ""),
-            (4, "RenderTargetArrayIndex", ""),
-            (5, "ViewPortArrayIndex", ""),
-            (6, "ClipDistance", ""),
-            (7, "CullDistance", ""),
-            (8, "OutputControlPointID", ""),
-            (9, "DomainLocation", ""),
-            (10, "PrimitiveID", ""),
-            (11, "GSInstanceID", ""),
-            (12, "SampleIndex", ""),
-            (13, "IsFrontFace", ""),
-            (14, "Coverage", ""),
-            (15, "InnerCoverage", ""),
-            (16, "Target", ""),
-            (17, "Depth", ""),
-            (18, "DepthLessEqual", ""),
-            (19, "DepthGreaterEqual", ""),
-            (20, "StencilRef", ""),
-            (21, "DispatchThreadID", ""),
-            (22, "GroupID", ""),
-            (23, "GroupIndex", ""),
-            (24, "GroupThreadID", ""),
-            (25, "TessFactor", ""),
-            (26, "InsideTessFactor", ""),
-            (27, "ViewID", ""),
-            (28, "Barycentrics", ""),
-            (29, "ShadingRate", ""),
-            (30, "CullPrimitive", ""),
-            (31, "StartVertexLocation", ""),
-            (32, "StartInstanceLocation", ""),
-            (33, "Invalid", ""),
-            ])
+        SemanticKind = db_dxil_enum(
+            "SemanticKind",
+            "Semantic kind; Arbitrary or specific system value.",
+            [
+                (0, "Arbitrary", ""),
+                (1, "VertexID", ""),
+                (2, "InstanceID", ""),
+                (3, "Position", ""),
+                (4, "RenderTargetArrayIndex", ""),
+                (5, "ViewPortArrayIndex", ""),
+                (6, "ClipDistance", ""),
+                (7, "CullDistance", ""),
+                (8, "OutputControlPointID", ""),
+                (9, "DomainLocation", ""),
+                (10, "PrimitiveID", ""),
+                (11, "GSInstanceID", ""),
+                (12, "SampleIndex", ""),
+                (13, "IsFrontFace", ""),
+                (14, "Coverage", ""),
+                (15, "InnerCoverage", ""),
+                (16, "Target", ""),
+                (17, "Depth", ""),
+                (18, "DepthLessEqual", ""),
+                (19, "DepthGreaterEqual", ""),
+                (20, "StencilRef", ""),
+                (21, "DispatchThreadID", ""),
+                (22, "GroupID", ""),
+                (23, "GroupIndex", ""),
+                (24, "GroupThreadID", ""),
+                (25, "TessFactor", ""),
+                (26, "InsideTessFactor", ""),
+                (27, "ViewID", ""),
+                (28, "Barycentrics", ""),
+                (29, "ShadingRate", ""),
+                (30, "CullPrimitive", ""),
+                (31, "StartVertexLocation", ""),
+                (32, "StartInstanceLocation", ""),
+                (33, "Invalid", ""),
+            ],
+        )
         self.enums.append(SemanticKind)
         SigPointKind = db_dxil_enum("SigPointKind", "Signature Point is more specific than shader stage or signature as it is unique in both stage and item dimensionality or frequency.", [
             (0, "VSIn", "Ordinary Vertex Shader input from Input Assembler"),
             (1, "VSOut", "Ordinary Vertex Shader output that may feed Rasterizer"),
             (2, "PCIn", "Patch Constant function non-patch inputs"),